theano/gradient.py

```diff
@@ -1606,12 +1608,13 @@ def verify_grad(fun, pt, n_tests=2, rng=None, eps=None,
     pt = [numpy.array(p) for p in pt]
 
     for i, p in enumerate(pt):
-        if p.dtype not in ('float32', 'float64'):
```

Comment:
line 1575 1592 need update in the docstring. For 1592, maybe just tell that it don't work for float16 for now?